content/how-to/lazy-load/react.md

```diff
@@ -0,0 +1,148 @@
+
+As a component, you can also make sure your own dependencies are lazily loaded. This is useful if a component relies on a really heavy library. Let's say we have a Todo component that optionally supports code highlighting...
+
+```jsx
+class Todo extends React.Component {
+  render() {
+    return (
+      <div>
+        {this.props.isCode ? <Highlight>{content}</Highlight> : content}
+      </div>
+    );
+  }
+}
+```
+
+We could then make sure the expensive library powering the Highlight component is only loaded when we actually want to highlight some code:
+
+```jsx
+// Highlight.js
+class Highlight extends React.Component {
+  render() {
+    const {Highlight} = this.props.highlight;
+    // highlight js is now on our props for use
+  }
+}
+export LazilyLoadFactory(Highlight, {
+  highlight: require('bundle?lazy!highlight'),
+});
+```
+
+Notice how the consumer of Highlight component had no idea it had a dependency that was lazily loaded? Or that if a user had todos with no code we would never need to load highlight.js? Also, we didn't need to specify a name for the highlight.js library because we don't care if webpack gives it a bundle.
+
+
+
+## The Code
+
+Source code of the LazilyLoad component module which exports both the Component interface and the higher order component interface, as well as the importLazy function to make ES2015 defaults feel a bit more natural.
+
+```jsx
+import React from 'react';
+
+const toPromise = (load) => (new Promise((resolve) => (
+  load(resolve)
+)));
+
+class LazilyLoad extends React.Component {
+
+  constructor() {
+    super(...arguments);
+    this.state = {
+      isLoaded: false,
+    };
+  }
+
+  componentWillMount() {
+    this.load(this.props);
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillReceiveProps(next) {
+    if (next.modules === this.props.modules) return null;
+    this.load(next);
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  load(props) {
+    this.setState({
+      isLoaded: false,
+    });
+
+    const { modules } = props;
+    const keys = Object.keys(modules);
+
+    Promise.all(keys.map((key) => toPromise(modules[key])))
+      .then((values) => (keys.reduce((agg, key, index) => {
+        agg[key] = values[index];
+        return agg;
+      }, {})))
+      .then((result) => {
+        if (!this._isMounted) return null;
+        this.setState({ modules: result, isLoaded: true });
+      });
+  }
+
+  render() {
+    if (!this.state.isLoaded) return null;
+    return React.Children.only(this.props.children(this.state.modules));
+  }
+}
+
+LazilyLoad.propTypes = {
+  children: React.PropTypes.func.isRequired,
+};
+
+export const LazilyLoadFactory = (Component, modules) => {
+  return (props) => (
+    <LazilyLoad modules={modules}>
+      {(mods) => <Component {...mods} {...props} />}
+    </LazilyLoad>
+  );
+};
+
+export const importLazy = (fetch) => (cb) => (
+   fetch((mod) => cb(mod.default))
```

Comment:
Instead of fetch let's use `System.import()`, and then that way you dont _have to use_ bundle-loader.

Would you consider showing both ways. Super valuable. webpack will return a Promise the same way so not much code changes.